File: TMessagesProj/src/main/java/org/telegram/android/NotificationRepeat.java
/*
 * This is the source code of Telegram for Android v. 1.7.x.
 * It is licensed under GNU GPL v. 2 or later.
 * You should have received a copy of the license in this archive (see LICENSE).
 *
 * Copyright Nikolai Kudashov, 2013-2014.
 */

package org.telegram.android;

import android.app.IntentService;
import android.content.Intent;

import org.telegram.messenger.Utilities;

public class NotificationRepeat extends IntentService {

    public NotificationRepeat() {
        super("NotificationRepeat");
    }

    @Override
    protected void onHandleIntent(Intent intent) {
        AndroidUtilities.runOnUIThread(new Runnable() {
            @Override
            public void run() {
                NotificationsController.getInstance().repeatNotificationMaybe();
            }
        });
    }
}

File: TMessagesProj/src/main/java/org/telegram/messenger/NotificationRepeat.kt
/*
 * This is the source code of Telegram for Android v. 5.x.x.
 * It is licensed under GNU GPL v. 2 or later.
 * You should have received a copy of the license in this archive (see LICENSE).
 *
 * Copyright Nikolai Kudashov, 2013-2018.
 */
package org.telegram.messenger

import android.app.IntentService
import android.content.Intent

class NotificationRepeat : IntentService("NotificationRepeat") {
    override fun onHandleIntent(intent: Intent?) {
        if (intent == null) {
            return
        }
        val currentAccount = intent.getIntExtra("currentAccount", UserConfig.selectedAccount)
        if (!UserConfig.isValidAccount(currentAccount)) {
            return
        }
        AndroidUtilities.runOnUIThread {
            NotificationsController.getInstance(currentAccount).repeatNotificationMaybe()
        }
    }
}